brightwind/transform/transform.py

```diff
@@ -252,7 +275,7 @@ def _round_timestamp_down_to_averaging_prd(timestamp, period):
                                                                 day=timestamp.day, hour=timestamp.hour,
                                                                 minute=_round_down_to_multiple(timestamp.minute,
                                                                                                int(period[:-3])))
-    elif period[-1] == 'h':
+    elif period[-1] == 'h' or period[-1] == 'H':
```

Comment:
Hi @olivia-bentley,

This edit triggered me to look up the chain as to what calls this function as there is no depreciation warning. It goes up to `merge_datasets_by_period()`.

When that is called, the `_freq_str_to_dateoffset()`, where we handle the backward compatibility, isn't called in the chain down to this `_round_timestamp_down_to_averaging_prd()`. Therefore, I'm not sure that the `merge_datasets_by_period()` handles backward compatibility and would break. Later in the function it calls `average_data_by_period()` which does hit `_freq_str_to_dateoffset()` but it would probably fail before it gets there.

Could you investigate if this is the case?

Thanks,

Reply:
I think searching for the word "period" in the public functions and seeing if the period argument passes through `_freq_str_to_dateoffset()` would be the thing to do to ensure we haven't missed any.